unit-test/sc_loads_tests.c

```diff
@@ -40,368 +40,404 @@
+     * for a cmd message to follow it, but it has a nonzero command number
+     */
+    InvalidEntry->CmdNumber = LastValidEntry->CmdNumber + 1;
 
     /* Execute the function being tested */
     SC_LoadAts(AtsIndex);
 
     /* Verify results */
-    UtAssert_True(SC_AppData.AtsCmdIndexBuffer[AtsIndex][0] == SC_ERROR,
-                  "SC_AppData.AtsCmdIndexBuffer[AtsIndex][0] == SC_ERROR");
-    UtAssert_True(SC_OperData.AtsCmdStatusTblAddr[AtsIndex][0] == SC_EMPTY,
-                  "SC_OperData.AtsCmdStatusTblAddr[AtsIndex][0] == SC_EMPTY");
-    UtAssert_True(SC_OperData.AtsInfoTblAddr[AtsIndex].NumberOfCommands == 0,
-                  "SC_OperData.AtsInfoTblAddr[AtsIndex].NumberOfCommands == 0");
-    UtAssert_True(SC_OperData.AtsInfoTblAddr[AtsIndex].AtsSize == 0,
-                  "SC_OperData.AtsInfoTblAddr[AtsIndex].AtsSize == 0");
-
+    UtAssert_INT32_EQ(SC_AppData.AtsCmdIndexBuffer[AtsIndex][0], SC_ERROR);
+    UtAssert_UINT32_EQ(SC_OperData.AtsCmdStatusTblAddr[AtsIndex][0], SC_EMPTY);
+    UtAssert_ZERO(SC_OperData.AtsInfoTblAddr[AtsIndex].NumberOfCommands);
+    UtAssert_ZERO(SC_OperData.AtsInfoTblAddr[AtsIndex].AtsSize);
     UtAssert_STUB_COUNT(CFE_EVS_SendEvent, 0);
 }
 
-void SC_LoadAts_Test_AtsEntryOverflow(void)
+void SC_LoadAts_Test_AtsEmpty(void)
 {
-    SC_AtsEntryHeader_t *Entry;
-    uint8                AtsIndex = 0;
-    size_t               MsgSize1;
-    size_t               MsgSize2;
-    int                  BufEntrySize;
-    int                  MaxBufEntries;
-    int                  i;
-    int                  j;
-
-    MsgSize1      = SC_PACKET_MAX_SIZE;
-    BufEntrySize  = ((MsgSize1 + SC_ROUND_UP_BYTES) / SC_BYTES_IN_WORD) + SC_ATS_HDR_NOPKT_WORDS;
-    MaxBufEntries = SC_ATS_BUFF_SIZE32 / BufEntrySize;
-
-    for (i = 0, j = 0; i < MaxBufEntries; i++, j += BufEntrySize)
-    {
-        Entry            = (SC_AtsEntryHeader_t *)&SC_OperData.AtsTblAddr[AtsIndex][j];
-        Entry->CmdNumber = i + 1;
-        UT_SetDataBuffer(UT_KEY(CFE_MSG_GetSize), &MsgSize1, sizeof(MsgSize1), false);
-    }
-
-    /* Next entry should not leave enough buffer space for an ATS command header */
-    Entry            = (SC_AtsEntryHeader_t *)&SC_OperData.AtsTblAddr[AtsIndex][j];
-    Entry->CmdNumber = i++ + 1;
-
-    /* Use the remaining buffer space to calculate the final message size */
-    MsgSize2 = (SC_ATS_BUFF_SIZE32 - SC_ATS_HDR_WORDS + 4 - j) * SC_BYTES_IN_WORD;
-    UT_SetDataBuffer(UT_KEY(CFE_MSG_GetSize), &MsgSize2, sizeof(MsgSize2), false);
+    uint8 AtsIndex = 0;
 
-    /* Set up final entry that will create condition */
-    j += ((MsgSize2 + SC_ROUND_UP_BYTES) / SC_BYTES_IN_WORD) + SC_ATS_HDR_WORDS;
-    Entry            = (SC_AtsEntryHeader_t *)&SC_OperData.AtsTblAddr[AtsIndex][j];
-    Entry->CmdNumber = i + 1;
-    UT_SetDataBuffer(UT_KEY(CFE_MSG_GetSize), &MsgSize1, sizeof(MsgSize1), false);
+    UT_SC_GetAtsTable(AtsIndex);
```

Comment:
What are the calls to UT_SC_GetAtsTable for when you don't use the returned value?  Looks like a repeated pattern in a handful of tests.

Reply:
Thanks for catching that, you are correct this has no effect after the rebase.  Removing these calls.